sdk/src/io.rs

```diff
@@ -140,45 +142,41 @@ pub mod proof_serde {
         if deserializer.is_human_readable() {
             let hex_bytes = String::deserialize(deserializer).unwrap();
             let bytes = hex::decode(hex_bytes).unwrap();
-            let proof = bincode::deserialize(&bytes).map_err(serde::de::Error::custom)?;
-            Ok(proof)
+            let mut de = rmp_decode::Deserializer::new(&bytes[..]);
+            Proof::<SC>::deserialize(&mut de).map_err(serde::de::Error::custom)
         } else {
             Proof::<SC>::deserialize(deserializer)
         }
     }
 
     #[cfg(test)]
     mod tests {
-
-        use crate::{
-            utils::{setup_logger, BabyBearPoseidon2},
-            SP1ProofWithIO, SP1Prover, SP1Stdin, SP1Verifier,
-        };
+        use crate::{SP1ProofWithIO, SP1Prover, SP1Stdin, SP1Verifier};
 
         pub const FIBONACCI_IO_ELF: &[u8] =
             include_bytes!("../../examples/fibonacci-io/program/elf/riscv32im-succinct-zkvm-elf");
 
-        /// Tests serialization with a human-readable encoding
+        /// Tests serialization with a human-readable encoding (JSON)
         #[test]
         fn test_json_roundtrip() {
             let mut stdin = SP1Stdin::new();
             stdin.write(&3u32);
             let proof = SP1Prover::prove(FIBONACCI_IO_ELF, stdin).unwrap();
             let json = serde_json::to_string(&proof).unwrap();
-            let output = serde_json::from_str::<SP1ProofWithIO<BabyBearPoseidon2>>(&json).unwrap();
+            let output = serde_json::from_str::<SP1ProofWithIO<_>>(&json).unwrap();
             SP1Verifier::verify(FIBONACCI_IO_ELF, &output).unwrap();
         }
 
-        /// Tests serialization with a binary encoding
+        /// Tests serialization with MsgPack encoding
```

Comment:
we should also have a bincode test

Reply:
I removed the dependency though. Why is bincode important?

Reply:
updated to leave it